cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -720,8 +723,10 @@ public ApplicationWithPrograms deployAppAndArtifact(NamespaceId namespace, @Null
 
     ArtifactDetail artifactDetail = artifactRepository.addArtifact(artifactId, jarFile);
     try {
-      return deployApp(namespace, appName, null, configStr, null, programTerminator, artifactDetail, ownerPrincipal,
-                       updateSchedules, false, Collections.emptyMap());
+      RunId runId = RunIds.generate();
+      String appVersion = RunIds.getTime(runId, TimeUnit.MILLISECONDS) + runId.getId();
```

Comment:
yes that is true. We did have a discussion on putting these api changes behind a feature flag - however, we went back on it since it would introduce a lot of tech debt. Looping in @avirmani @chtyim for visibility

Reply:
discussed offline

Reply:
These are internal APIs. What is the specific backward compat concerns, will this return extra fields in json? Also we have release as preview, does that alleviate concerns?

Reply:
The apis will operate on the "latest version" of the app (which is generated when we deploy an app), currently most of the apis operate on default -SNAPSHOT version, however we change that behavior and map -SNAPSHOT to latest - that shouldn't necessarily break the user and helps with the backward compatibility issue. This is what we discussed offline